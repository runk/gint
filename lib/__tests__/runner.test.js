'use strict';

const path = require('path');
const rimraf = require('rimraf');
const tempy = require('tempy');
const runner = require('../runner');
const { write } = require('../../__tests__/helpers');

describe('lib/runner', () => {
  let dir;
  beforeEach(() => { dir = tempy.directory(); });
  afterEach(() => rimraf.sync(dir));

  describe('strict config', () => {
    const config = {
      rules: {
        'commit-min-len': ['error', 10],
        'commit-max-len': ['warn', 512]
      }
    };

    it('should return correct output - no errors', () =>
      runner(config, { hookName: 'commit-msg', hookParams: 'Fix all the things' })
        .then((output) => expect(output).toBe(''))
    );

    it('should return correct output - errors', () =>
      runner(config, { hookName: 'commit-msg', hookParams: 'Fix' })
        .catch((output) => expect(output).toMatchSnapshot())
    );

    it('should return correct output - no errors when no applicable rules for hook', () =>
      runner(config, { hookName: 'pre-commit', hookParams: 'Fix' })
        .catch((output) => expect(output).toBe(''))
    );
  });

  describe('silent config', () => {
    const config = {
      rules: {
        'commit-min-len': ['ignore', 10],
      }
    };

    it('should return correct output - no errors', () =>
      runner(config, { hookName: 'commit-msg', hookParams: 'Fix all the things' })
        .then((output) => expect(output).toBe(''))
    );

    it('should return correct output - errors', () =>
      runner(config, { hookName: 'commit-msg', hookParams: 'Fix' })
        .catch((output) => expect(output).toBe(''))
    );
  });

  describe('non-strict config', () => {
    const config = {
      rules: {
        'commit-min-len': ['warn', 10],
        'commit-max-len': ['warn', 512],
        "commit-message-pattern": ["warning", "^\\[(patch|minor|major)\\]\\s.{3,}", "i"],
      }
    };

    it('should return correct output - no errors', () =>
      runner(config, { hookName: 'commit-msg', hookParams: '[patch] Fix all the things' })
        .then((output) => expect(output).toBe(''))
    );

    it('should return correct output - errors', () =>
      runner(config, { hookName: 'commit-msg', hookParams: 'Fix' })
        .then((output) => expect(output).toMatchSnapshot())
    );
  });
});
