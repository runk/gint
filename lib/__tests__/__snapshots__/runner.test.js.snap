// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/runner non-strict config should return correct output - errors 1`] = `
"Gint found some issues:
  [[33mwarning[39m]	[90mcommit-min-len[39m	Commit message is too short, min length: 10 chars
  [[33mwarning[39m]	[90mcommit-message-pattern[39m	Commit message does not match pattern \\"^\\\\[(patch|minor|major)\\\\]\\\\s.{3,}\\"

[31m[1m âœ– 2 problems (0 errors, 2 warnings)[22m[39m
"
`;

exports[`lib/runner strict config should return correct output - errors 1`] = `
"Gint found some issues:
  [[31merror[39m]	[90mcommit-min-len[39m	Commit message is too short, min length: 10 chars

[31m[1m âœ– 1 problem (1 error, 0 warnings)[22m[39m
"
`;
